sys/net/gnrc/link_layer/netdev2/gnrc_netdev2_ieee802154.c

```diff
@@ -211,9 +249,79 @@ static int _send(gnrc_netdev2_t *gnrc_netdev2, gnrc_pktsnip_t *pkt)
     else {
         return -ENOBUFS;
     }
+#ifdef MODULE_NETDEV_RETRANS
+    /* queue the packet for potential retransmissions */
+    if (gnrc_netdev2->retrans_head->pkt != pkt) {
+        netdev2_retrans_queue_t *pkt_node = _alloc_pkt_node(pkt);
+        if (pkt_node == NULL) {
+            DEBUG("_send_ieee802154: could not add packet to packet queue\n");
+        }
+        else {
+            gnrc_pktqueue_add((gnrc_pktqueue_t**) &(gnrc_netdev2->retrans_head), (gnrc_pktqueue_t*) pkt_node);
+        }
+    }
+    gnrc_pktbuf_hold(pkt, 1);
+#endif
     /* release old data */
     gnrc_pktbuf_release(pkt);
     return res;
 }
 
+static int _set(gnrc_netdev2_t *gnrc_netdev2, netopt_t opt, void *val, size_t len)
+{
+    int res = 0;
+    netdev2_ieee802154_t *dev = (netdev2_ieee802154_t *)gnrc_netdev2->dev;
+
+    if (dev == NULL) {
+        return -ENODEV;
+    }
+
+#ifdef MODULE_NETDEV_RETRANS
+    if (opt == NETOPT_RETRANS) {
+        if (len > sizeof(uint8_t)) {
```

Comment:
There is no doc on the type of that value. Are you sure assuming `uint8_t` is okay? Usually unsigned values in netopt are `uint16_t`.